### PR TITLE
[DJ][12]: Added categories component for home page

### PR DESCRIPTION
**Pull request by purple-mountain**
image

### CODE REVIEW

**Review by Ismoil-JS — APPROVED** [review]

File: src/modules/home/components/category/category-item.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import Typography from '@mui/material/Typography';
+import { type CategoryWithIcon } from '../../types/category-with-icon.type';
+import Box from '@mui/material/Box';
+
+const categoryItemContainerStyles = {
```
Comment (MdRasulov) [inline src/modules/home/components/category/category-item.tsx:5]: Let's move the styles to the separate file. You can create another file in the same folder with the name: _filename_.styles.ts. Inside it create an object like so:


```ts
import { MuiStylesObject } from '@src/types/utility.types';

export const categoryStyles = {
     categoryItemContainer: {
       // write you styles here
     }

} satisfies MuiStylesObject
```

of course you will need to create a custom type for that first:

```ts
import { SxProps, Theme } from '@mui/material';

export type MuiStylesObject = Record<string, SxProps<Theme>>;
```

File: src/modules/home/components/category/category-item.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  display: 'flex',
+  flexDirection: 'column',
+  justifyContent: 'center',
+  alignItems: 'center',
+  paddingX: 4,
+  color: 'gray',
+  paddingY: 2,
+  '&:hover': {
+    backgroundColor: 'rgba(0, 0, 0, 0.04)',
+    color: '#000',
```
Comment (MdRasulov) [inline src/modules/home/components/category/category-item.tsx:15]: Avoid just color codes. If its commonly used color, create a constant with the name for that color and use that.

File: src/modules/home/components/category/category-list.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
+import Box from '@mui/material/Box';
+import { CategoryItem } from './category-item';
+import { CategoryScrollArrow } from './category-scroll-arrow';
+import { categories } from '../../data/category-data';
+import { useNavigate, useLocation } from 'react-router-dom';
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+const containerStyles = {
```
Comment (MdRasulov) [inline src/modules/home/components/category/category-list.tsx:8]: Move styles.

File: src/modules/home/routes/home.route.tsx [hunk 4]
```diff
@@ -1,12 +1,32 @@
 import Box from '@mui/material/Box';
 import { HeaderComponent } from '../components/header';
 import { PropertyList } from '../components/property';
+import Container from '@mui/material/Container';
+import Divider from '@mui/material/Divider';
+import { CategoryList } from '../components/category/category-list';
+
+const containerStyles = { display: 'flex', flexDirection: 'column', width: '100%' };
+
+const dividerStyles = {
```
Comment (MdRasulov) [inline src/modules/home/routes/home.route.tsx:10]: The same here with the styles and usage

**Review by MdRasulov — CHANGES_REQUESTED** [review]
Good job! Left some comments

**Pull request merged**